climpred/smoothing.py

```diff
@@ -0,0 +1,248 @@
+            (optional)
+         method (str): Regridding method. Options are:
+            - 'bilinear'
+            - 'conservative', **need grid corner information**
+            - 'patch'
+            - 'nearest_s2d'
+            - 'nearest_d2s'
+         periodic (bool): Periodic in longitude? Default to False. optional
+            Only useful for global grids with non-conservative regridding.
+            Will be forced to False for conservative regridding.
+         filename (str): Name for the weight file. (optional)
+            The default naming scheme is:
+                 {method}_{Ny_in}x{Nx_in}_{Ny_out}x{Nx_out}.nc
+                 e.g. bilinear_400x600_300x400.nc
+         reuse_weights (bool) Whether to read existing weight file to save
+            computing time. False by default. (optional)
+
+        Returns:
+            ds (xarray.object) regridded
+    """
+
+    if xe is None:
+        raise ImportError(
+            'xesmf is not installed; see'
+            'https://xesmf.readthedocs.io/en/latest/installation.html'
+        )
+
+    def _regrid_it(da, d_lon, d_lat, **kwargs):
+        """
+        Global 2D rectilinear grid centers and bounds
+
+        Args:
+            da (xarray.DataArray): Contain input and output grid coords.
+                Look for variables ``lon``, ``lat``, ``lon_b``, ``lat_b`` for
+                conservative method, and ``TLAT``, ``TLON`` for CESM POP grid
+                Shape can be 1D (Nlon,) and (Nlat,) for rectilinear grids,
+                or 2D (Ny, Nx) for general curvilinear grids.
+                Shape of bounds should be (N+1,) or (Ny+1, Nx+1).
+            d_lon (float): Longitude step size, i.e. grid resolution
+            d_lat (float): Latitude step size, i.e. grid resolution
+        Returns:
+            da : xarray DataArray with coordinate values
+        """
+
+        def check_lon_lat_present(da):
+            if 'lat' in ds.coords and 'lon' in ds.coords:
+                return da
+            elif 'lat_b' in ds.coords and 'lon_b' in ds.coords:
+                return da
+            # for CESM POP grid
+            elif 'TLAT' in ds.coords and 'TLONG' in ds.coords:
+                da = da.rename({'TLAT': 'lat', 'TLONG': 'lon'})
+                return da
+            else:
+                raise ValueError(
+                    'lon/lat or lon_b/lat_b or TLAT/TLON not found, please rename.'
+                )
+
+        da = check_lon_lat_present(da)
+        grid_out = {
+            'lon': np.arange(da.lon.min(), da.lon.max() + d_lon, d_lon),
+            'lat': np.arange(da.lat.min(), da.lat.max() + d_lat, d_lat),
+        }
+        regridder = xe.Regridder(da, grid_out, **kwargs)
+        return regridder(da)
+
+    if 'lon' not in d_lon_lat_dict:
+        d_lon_lat_dict['lon'] = d_lon_lat_dict['lat']
+    elif 'lat' not in d_lon_lat_dict:
+        d_lon_lat_dict['lat'] = d_lon_lat_dict['lon']
+    else:
```

Comment:
This doesn't work. It executes with the default keyword, since the if statements just rely on only missing one of lat or lon. You should just do `if ('lon' not in d_lon_lat_dict) and ('lat' not in d_lon_lat_dict)` or some similar setup.

Reply:
made it more explicit